### ISSUE TITLE
ZIP function seems to set fixed length key name

### ISSUE THREAD

**Issue by Soultripper** [arangodb/arangodb#1746]
I seem to now have an issue with the ZIP function based on latest updates.

Querying an edge collection, which has a property called type, it seems the string value is a fixed length, and gets padded out

```
LET actions = (FOR user, edge IN INBOUND film actioned
FILTER edge._from == @userId
COLLECT toFilm = edge._to INTO myActions
RETURN ZIP(myActions[*].edge.type, myActions[*].edge.createdAt))
```

Results 
```
{
"watched": 1455569572715,
"ownched": 1455569575342,
"loveded": 1455569574393
}

```
however, this should read
```

{
"watched": 1455569572715,
"own": 1455569575342,
"loved": 1455569574393
}

```
I've checked the edge 'type' property in the collection and is correct:

```
{
"createdAt": 1455569574393,
"type": "loved"
} 
```
so it's corrupted by the recent ZIP call.

Any ideas?

**Comment by CoDEmanX** [comment]
Does your server include the following commit?

https://github.com/arangodb/arangodb/commit/b8849daa508c7db413ea03f34c4f75476e3d19d3

**Comment by Soultripper** [comment]
Hey

I've just updated to 2.8.2 from 2.8.0 (using the APP version) but seems to have same problem. 

If I find time soon, will compile from source ensuring the above change is committed

What's interesting is that it doesn't happen on every record. I have another film which returns just fine:

The _key of the two films are:

works:    _"films/billy-elliot-2000"_
breaks: _"films/cl-o-from-5-to-7-1962"_

There could be an issue with the char set as the raw title of the breaking film : **Cléo from 5 to **7

**Comment by CoDEmanX** [comment]
2.8.2 does neither contain the invalid fix nor the fix of the invalid fix, so I guess it will be fixed in 2.8.3. Would be great if you could confirm if it's working with the latest devel version.

**Comment by dothebart** [comment]
2.8.3 has been released yesterday. Can you revalidate @Soultripper ?
